### PR TITLE
Return proper error string if sharing for this user is disabled

### LINKED ISSUE #22402: Different errors when you select the "Exclude groups to share " on v8.2.2 and v9.0
### Steps to reproduce
1. Install the Desktop client v 2.2.0 with Server v 9.0, with yellow user
2. Go to the Server Web
3. Select the option "Exclude groups from sharing"
screen shot 2016-02-15 at 15 13 24
4. Go to the Desktop client
5. Create a new folder
6. Share with owncloud
7. Write on 'Share with users or gruops', the group that include the yellow user, in this case colors

### Actual behaviour
When you select the group (e.g. colors), appears this error: 
screen shot 2016-02-15 at 14 56 51
LOG
```
02-15 14:57:39:355 0x7fb6db521fe0 OCC::OcsJob::finished: Reply to "POST"  QUrl( "https://xxxxxxxxxxx/ocs/v1.php/apps/files_sharing/api/v1/shares" )  (QPair("path","/Share_with_others"), QPair("shareType","1"), QPair("shareWith","colors")) has unexpected status code: 403 "{"ocs":{"meta":{"status":"failure","statuscode":403,"message":"The Share API is disabled"},"data":[]}}"
02-15 14:57:39:356 0x7fb6db521fe0 OCC::ShareUserGroupWidget::displayError: Error from server 403 "The Share API is disabled"
```

NOTE: If I do the same with a server 8.2.2, I have the different error:
screen shot 2016-02-15 at 14 59 22

### Server configuration
Desktop v ownCloud-2.2.0.3123-nightly20160215.pkg
Server v {"installed":true,"maintenance":false,"version":"8.2.2.2","versionstring":"8.2.2","edition":"Enterprise"}
{"installed":true,"maintenance":false,"version":"9.0.0.11","versionstring":"9.0 beta 1","edition":""}

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #22402

Makes the error message if a user is not allowed to share a bit better.

@Dianafg76 please test.

### CODE REVIEW

File: lib/private/share20/manager.php [hunk 1]
```diff
@@ -455,18 +455,16 @@ protected function pathCreateChecks($path) {
 	 * Check if the user that is sharing can actually share
 	 *
 	 * @param \OCP\Share\IShare $share
-	 * @return bool
+	 * @throws \Exception
 	 */
 	protected function canShare(\OCP\Share\IShare $share) {
 		if (!$this->shareApiEnabled()) {
-			return false;
+			throw new \Exception('The share API is disabled');
```
Comment (nickvergessen) [inline lib/private/share20/manager.php]: Where is this used, should it be a hintException with translation?

**Comment by nickvergessen** [comment]
:+1: otherwise

**Comment by Dianafg76** [comment]
:+1:

File: lib/private/share20/manager.php (hunk 1 above)
Comment (rullzer) [inline lib/private/share20/manager.php]: It is basically thrown in the OCS Share API...
I want to use proper hint exceptions as well. But right now that messes with the status codes to much.

I'll create an issue so I don't forget.

**Pull request merged**